ci-operator/step-registry/upi/conf/vsphere/dns/upi-conf-vsphere-dns-commands.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if command -v pwsh &> /dev/null
+then
+  echo "vmc-ci.devcluster.openshift.com" > "${SHARED_DIR}"/basedomain.txt
+
+  cluster_name=${NAMESPACE}-${UNIQUE_HASH}
+  base_domain=$(<"${SHARED_DIR}"/basedomain.txt)
+  cluster_domain="${cluster_name}.${base_domain}"
+
+  export AWS_DEFAULT_REGION=us-west-2  # TODO: Derive this?
+  export AWS_SHARED_CREDENTIALS_FILE=/var/run/vault/vsphere/.awscred
+  export AWS_MAX_ATTEMPTS=50
+  export AWS_RETRY_MODE=adaptive
+  export HOME=/tmp
+
+  if ! command -v aws &> /dev/null
+  then
+      echo "$(date -u --rfc-3339=seconds) - Install AWS cli..."
+      export PATH="${HOME}/.local/bin:${PATH}"
+      if command -v pip3 &> /dev/null
+      then
+          pip3 install --user awscli
+      else
+          if [ "$(python -c 'import sys;print(sys.version_info.major)')" -eq 2 ]
+          then
+            easy_install --user 'pip<21'
+            pip install --user awscli
+          else
+            echo "$(date -u --rfc-3339=seconds) - No pip available exiting..."
+            exit 1
+          fi
+      fi
+  fi
+
+  # Load array created in setup-vips:
+  # 0: API
+  # 1: Ingress
+  declare -a vips
+  mapfile -t vips < "${SHARED_DIR}"/vips.txt
```

Comment:
Consider checking for the existence of `${SHARED_DIR}/vips.txt `, otherwise `api_dns_target`  and `apps_dns_target` may contain invalid content.

Reply:
I can add a check here to see if the file exists and if not, also exit out w/ error.  I do see this as being helpful if used by new workflow and someone forgetting to set it.  vips.txt should be set by previous step.  I would expect that step to fail out if any issues were to occur, but do agree this would be great to have.